core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala

```diff
@@ -83,6 +83,8 @@ class CheckpointFile[T](val file: File, version: Int, formatter: CheckpointFileF
         if (line == null)
           return Seq.empty
         line.toInt match {
+          // Backticks are required to match against the value of the version field instead of introducing a new
+          // variable called version
```

Comment:
@junrao That doesn't quite work. The idea is that if you want to match on the value of a variable, it has to start with uppercase or backticks are required. As you say, this is not too clear for people who don't know though, so I pushed a new version that uses a guard instead. That should be clear and we no longer need a multi-line comment. Let me know if that's OK.